src/librustdoc/json/mod.rs

```diff
@@ -69,6 +69,20 @@ impl JsonRenderer<'tcx> {
                     .iter()
                     .filter_map(|i| {
                         let item = &i.impl_item;
+
+                        // HACK(hkmatsumoto): On impls for primitive types, we push the item
+                        // to the index even if it is not local. This is because we have to
+                        // index `std` when getting information of methods of primitive types
+                        // while many primitive types are originates from `core`, as a workaround
+                        // for the behavior of `ExternalCrate::primitives`.
```

Comment:
Alternatively, we can make a new function that retrieves not only top level items but all items, and use that instead of `ExternalCrate::primitives`. This way we can remove this part of change.

Reply:
I don't understand the suggestion nor the comment. What do top-level items have to do with it? Where is `primitives()` used? All primitives are currently documented at the top-level so checking all items shouldn't change anything.